scripts/batch-score.coffee

```diff
@@ -0,0 +1,107 @@
+# Description:
+#   Script for batch wise score.
+#
+# Configuration:
+#   INFO_SPREADSHEET_URL
+#
+# Commands:
+#   hubot score bxx
+# 
+# Author:
+#   aseem09
+
+module.exports = (robot) ->
+
+  # using this, stores the data in brain's "scorefield" key
+  scorefield = () ->
+    Field = robot.brain.get("scorefield") or {}
+    robot.brain.set("scorefield",Field)
+    Field
+  
+  # response for score status of any <batch>
+  stringLength = (str) ->
+    string=String(str)
+    temp=string.split('')
+    return temp.length
+  
+  pad = (array) ->
+    padding=0
+    max_length=0
+    for name in array
+      max_length= if max_length > stringLength(name) then max_length else stringLength(name)
+    padding = max_length
+    for i in [0..array.length-1]
+      max= padding- stringLength(array[i])
+      if i == 0
+        max=max+3
+      name=array[i]
+      if max>0
+        for j in [0..max-1]
+          name += " "
+      array[i]=name
+    return array
+  
+  robot.respond /score b(\d\d)/i, (msg) ->    
```

Comment:
I don't' know why do we need '('s around `\d`s. Would you care to explain?

Reply:
I tried. It doesnt work without the braces!